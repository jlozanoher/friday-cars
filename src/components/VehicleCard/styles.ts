import styled from "styled-components";

export const Title = styled.div`
  font-size: 20px;
  font-weight: 500;
  line-height: 24px;
`;

export const SubTitle = styled.div`
  font-size: 16px;
  line-height: 24px;
`;

export const Img = styled.img`
  width: 150px;
  margin: 8px;
`;
