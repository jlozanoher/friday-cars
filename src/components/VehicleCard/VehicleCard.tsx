import React from "react";
import Vehicle from "../../models/vehicle.model";
import { HStack, VStack } from "../Commons/Containers/Containers";
import CarIcon from "./car-icon.png";
import * as S from "./styles";
import cs from "classnames";

interface Props {
  vehicle: Vehicle;
  onClick?: (vehicle: Vehicle) => void;
  selected?: boolean;
}

const VehicleCard = (props: Props) => {
  const { vehicle: v, onClick = () => {}, selected } = props;

  const handleClick = () => onClick(v);

  return (
    <S.Card
      onClick={handleClick}
      className={cs({ selected: selected })}
      data-testid={`Vehicle${v.id}`}
    >
      <S.Img src={CarIcon} alt={"car-icon"} />
      <HStack>
        <S.Title>{v.make}</S.Title>
        <S.SubTitle>{v.model}</S.SubTitle>
      </HStack>
      <VStack>
        <div>
          <small>Body type: {v.bodyType}</small>
        </div>
        <div>
          <small>Engine capacity: {v.engineCapacity} cc</small>
        </div>
        <div>
          <small>Power (Horsepower): {v.enginePowerPS}</small>
        </div>
        <div>
          <small>Power (KiloWatts): {v.enginePowerPW || "-"}</small>
        </div>
        <div>
          <small>Fuel type: {v.fuelType}</small>
        </div>
      </VStack>
    </S.Card>
  );
};

export default VehicleCard;
