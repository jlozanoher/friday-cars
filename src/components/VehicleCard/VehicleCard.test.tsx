import renderer from "react-test-renderer";
import Vehicle from "../../models/vehicle.model";
import VehicleCard from "./VehicleCard";

const vehicle: Vehicle = {
  id: "1",
  make: "BMW",
  model: "3er",
  bodyType: "Kombi",
  engineCapacity: 1995,
  enginePowerPS: 190,
  enginePowerPW: 300,
  fuelType: "Diesel",
};

test("VehicleCard should match snap", () => {
  const tree = renderer.create(<VehicleCard vehicle={vehicle} />).toJSON();
  expect(tree).toMatchSnapshot();
});
