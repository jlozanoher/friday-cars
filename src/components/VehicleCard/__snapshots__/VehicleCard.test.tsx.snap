// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match snap 1`] = `
<div
  className="sc-gsDKAQ sc-dkPtRN sc-gKclnd kkVWFs hZpHlw jkCszs"
  data-testid="Vehicle1"
  onClick={[Function]}
>
  <img
    alt="car-icon"
    className="sc-jRQBWg flbqtC"
    src="car-icon.png"
  />
  <div
    className="sc-bdvvtL iuoSCa"
  >
    <div
      className="sc-hKwDye dMJKKO"
    >
      BMW
    </div>
    <div
      className="sc-eCImPb impPDT"
    >
      3er
    </div>
  </div>
  <div
    className="sc-gsDKAQ kkVWFs"
  >
    <div>
      <small>
        Body type: 
        Kombi
      </small>
    </div>
    <div>
      <small>
        Engine capacity: 
        1995
         cc
      </small>
    </div>
    <div>
      <small>
        Power (Horsepower): 
        190
      </small>
    </div>
    <div>
      <small>
        Power (KiloWatts): 
        300
      </small>
    </div>
    <div>
      <small>
        Fuel type: 
        Diesel
      </small>
    </div>
  </div>
</div>
`;
